File: eccorce_app/app/src/main/java/com/example/eccomerce_app/dto/StoreDto.kt
package com.example.eccomerce_app.dto

import com.example.hotel_mobile.services.kSerializeChanger.UUIDKserialize
import kotlinx.serialization.Serializable
import java.io.File
import java.util.UUID

@Serializable
data class StoreDto(
    @Serializable(with = UUIDKserialize::class)
    val id: UUID,
    @Serializable(with = UUIDKserialize::class)
    val userId: UUID,
    val name: String,
    val wallpaperImage: String,
    val smallImage: String,
    val longitude: Double,
    val latitude: Double,
)

@Serializable
data class StoreStatusDto(
    @Serializable(with = UUIDKserialize::class)
    val StoreId: UUID,
    val Status: Boolean
)

data class CreateStoreDto(
    val name: String?="سوق سعيد",
    val wallpaperImage: File?=null,
    val smallImage: File?=null,
    val longitude: Double?=null,
    val latitude: Double?=null
)

File: ecommerc_dotnet/Presentation/dto/StoreDto.cs
using System.ComponentModel.DataAnnotations;

namespace ecommerc_dotnet.Presentation.dto
{
    public class StoreDto
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string WallpaperImage  { get; set; }
        public string  SmallImage { get; set; }
        public string UserName { get; set; } = "";
        public bool  IsBlocked { get; set; }
        public decimal? Longitude { get; set; } = null;
        public decimal? Latitude { get; set; } = null;
        public Guid UserId { get; set; }
        public DateTime? UpdatedAtAt { get; set; } = null;
    }

    public class CreateStoreDto
    {
        [StringLength(maximumLength: 100, MinimumLength = 2, ErrorMessage = "Name must not be empty")]
        [Required]
        public string Name { get; set; } = string.Empty;

        [Required] public  IFormFile WallpaperImage { get; set; }
        [Required] public IFormFile SmallImage { get; set; }

        [Required]
        public decimal Longitude { get; set; }
        
        [Required]
        public decimal Latitude { get; set; }
    }
    
    
    public class UpdateStoreDto
    {
        [StringLength(maximumLength:100 ,ErrorMessage= "name must not  be empty")]
        public string? Name { get; set; }
        public IFormFile? WallpaperImage { get; set; } = null;
        public IFormFile? SmallImage { get; set; } = null;
        public decimal? Longitude  { get; set; }
        public decimal? Latitude    { get; set; }
        
       
    }
    
    public class StoreStatusDto
    {
        public  Guid StoreId { get; set; }
        public   bool Status { get; set; }
    }
}